pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java

```diff
@@ -480,22 +483,32 @@ public RebalanceResult rebalance(TableConfig tableConfig, Configuration rebalanc
     }
   }
 
-  private Map<InstancePartitionsType, InstancePartitions> getInstancePartitionsMap(TableConfig tableConfig,
-      boolean reassignInstances, boolean bootstrap, boolean dryRun) {
+  /**
+   * Gets the instance partitions for instance partition types and also returns a boolean for whether they are unchanged
+   */
+  private Pair<Map<InstancePartitionsType, InstancePartitions>, Boolean> getInstancePartitionsMap(
```

Comment:
(Optional) Might be more readable if we pass in a `MutableBoolean`

Reply:
How is this more readable?

Reply:
Currently the method name doesn't match the return value type, but I'm also fine with it